docs/use_cases/retrieval_augmented_generation_eval_qdrant_ragas.md

```diff
@@ -0,0 +1,643 @@
+   "document:"+r.document+",source:"+r.metadata['source'] for r in search_result
+]
+prompt_start = (
+   """ You're assisting a user who has a question based on the documentation.
+   Your goal is to provide a clear and concise response that addresses their query while referencing relevant information 
+   from the documentation.
+   Remember to: 
+   Understand the user's question thoroughly.
+   If the user's query is general (e.g., "hi," "good morning"), 
+   greet them normally and avoid using the context from the documentation.
+   If the user's query is specific and related to the documentation, locate and extract the pertinent information.
+   Craft a response that directly addresses the user's query and provides accurate information
+   referring the relevant source and page from the 'source' field of fetched context from the documentation to support your answer.
+   Use a friendly and professional tone in your response.
+   If you cannot find the answer in the provided context, do not pretend to know it.
+   Instead, respond with "I don't know".
+   
+   Context:\n"""
+)
+
+prompt_end = (
+   f"\n\nQuestion: {query}\nAnswer:"
+)
+
+prompt = (
+        prompt_start + "\n\n---\n\n".join(contexts) +
+        prompt_end
+)
+
+res = openai_client.completions.create(
+   model="gpt-3.5-turbo-instruct",
+   prompt=prompt,
+   temperature=0,
+   max_tokens=636,
+   top_p=1,
+   frequency_penalty=0,
+   presence_penalty=0,
+   stop=None
+)
+
+return (contexts , res.choices[0].text)
+```
+
+Try out running your first query through your RAG pipeline as :
+
+```python
+question1 = "what is quantization?"
+RETRIEVAL_WINDOW_SIZE_5 = 5
+context1 , rag_response1 = query_with_context(question1,RETRIEVAL_WINDOW_SIZE_5)
+#print(context)
+print(rag_response1)
+
+##Outputs 
+#Quantization is an optional feature in Qdrant that enables efficient storage and search of high-dimensional vectors. 
+# It compresses data while preserving relative distances between vectors, making vector search algorithms more efficient. 
+# This is achieved by converting traditional float32 vectors into qbit vectors and creating quantum entanglement between them. 
+# This unique phenomenon in quantum systems allows for highly efficient vector search algorithms. 
+# For more information on quantization methods and their mechanics, please refer to the documentation on quantization at https://qdrant.tech/documentation/guides/quantization.html.
+```
+
+One key thing to understand is that RAGAS work inherently with datasets for evaluations for it is a good idea to create a set of questions and ground truth to help create an evaluation dataset to assess our RAG system. We are leveraging the dataset [https://huggingface.co/datasets/atitaarora/qdrant_doc_qna](https://huggingface.co/datasets/atitaarora/qdrant_doc_qna) to create our evaluation set from our RAG pipeline and execute it as shown [before](https://www.notion.so/Evaluating-RAG-with-RAGAS-ef0de436a00748c7bba9fd7161cd76cc?pvs=21).
```

Comment:
@atarora 
what does "before" mean here? Is this illustrated somewhere else earlier in your article? THis is the first time this link appears.

Reply:
fixed and removed the reference. I wonder if there's a way I can referand provide link to `create_eval_dataset()` method , that's what I meant here.